apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceExtension.cs

```diff
@@ -78,6 +78,16 @@ public static class CloudTraceExtension
         internal const string TraceKey = "Google.Cloud.Diagnostics.AspNetCore.Trace";
 
         /// <summary>
+        /// A class to wrap the "shouldTraceFunc" param in <see cref="AddGoogleTrace"/>.
+        /// This is used instead of injecting the value of "shouldTraceFunc" directly 
+        /// to ensure we do not override a service the user setup.
+        /// </summary>
+        internal class ShouldTraceRequest
+        {
+            internal Func<HttpRequest, bool?> ShouldTrace { get; set; }
```

Comment:
Make this immutable, and a sealed class?
(It's annoying that it's needed. I wonder if there's some other way of marking things for dependency injection using attributes etc...)